src/createStore.js

```diff
@@ -238,15 +140,12 @@ export default function createStore(reducer, initialState, enhancer) {
     }
   }
 
-  // When a store is created, an "INIT" action is dispatched so that every
-  // reducer returns their initial state. This effectively populates
-  // the initial state tree.
-  dispatch({ type: ActionTypes.INIT })
+  replaceReducer(reducer)
 
   return {
     dispatch,
```

Comment:
Should we have the "real" store extend the store base here via spread, so that enhancers can add their own properties besides `dispatch` and `getState`?

```js
return {
  ...storeBase,
  dispatch,
  getState,
  subscribe,
  // ... and so on
}
```

E.g. this would allow the observable interface to be safely be implemented as an enhancer. Not that we'd necessarily want to do that — just an example of how alternative event systems are possible with this new enhancer architecture.